lib/game.rb

```diff
@@ -0,0 +1,142 @@
+class Game 
+
+  WIN_COMBINATIONS = [[0,1,2], [3,4,5], [6,7,8], [0,3,6], [1,4,7], [2,5,8], [0,4,8], [2,4,6]]
+
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize (player_1 = Player::Human.new("X"), player_2 = Player::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    @board.turn_count.even? ? player_1 : player_2
+  end
+
+  def over?
+    won? || draw? || board.full? ? true : false
+  end
+
+  def draw?
+    !won? && board.full? ? true : false
+  end
+
+  def full?
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |win_combo|
+      win_index_1 = win_combo[0]
+      win_index_2 = win_combo[1]
+      win_index_3 = win_combo[2]
+
+      position_1 = board.cells[win_index_1]
+      position_2 = board.cells[win_index_2]
+      position_3 = board.cells[win_index_3]
+      if position_1 == "X" && position_2 == "X" && position_3 == "X" || position_1 == "O" && position_2 == "O" && position_3 == "O"
+        return win_combo
+      end
+    end
+    return false
+  end
+
+  def winner
+    if won?
+      board.cells[won?[0]]
+    end
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.